quinn-proto/src/connection/mod.rs

```diff
@@ -6343,6 +6385,36 @@ impl Connection {
 
         Ok(probed_addresses)
     }
+
+    /// Attempts to continue a nat traversal round by trying to open paths for pending client probes.
+    ///
+    /// If there was nothing to do, it returns `None`. Otherwise it returns whether the path was
+    /// successfully open.
+    fn continue_nat_traversal_round(&mut self, now: Instant) -> Option<bool> {
+        let client_state = self.iroh_hp.client_side_mut().ok()?;
+        let (id, address) = client_state.continue_nat_traversal_round()?;
+        let ipv6 = self.paths.values().any(|p| p.data.remote.is_ipv6());
+        let open_result = self.open_nat_traversal_path(address, ipv6, now);
+        let client_state = self.iroh_hp.client_side_mut().expect("validated");
+        match open_result {
+            Ok(None) => Some(true),
+            Ok(Some((path_id, _remote, path_was_known))) => {
+                if !path_was_known {
+                    client_state.add_round_path_id(path_id);
+                }
+                Some(true)
+            }
+            Err(e) => {
+                if matches!(
```

Comment:
maybe this match check could be a function like `is_temporary_open_error` or sth like that, to avoid duplicating this logic here and in the other place and keep it better in sync